src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,84 @@
+package wolox.training.controllers;
+
+import javax.validation.constraints.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import wolox.training.exceptions.UserIdMismatchException;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+  @Autowired
+  private UserRepository userRepository;
+
+  @GetMapping
+  public Iterable findAll() {
+    return userRepository.findAll();
+  }
+
+  @GetMapping("/{id}")
+  public User findOne(@PathVariable Long id) {
+    try {
+      return userRepository.findById(id).orElseThrow(
```

Comment:
I don't like repeating this pattern a lot. Should I do like I did for the other findById and extract it to a method or is there something better I'm not seeing?

Reply:
You mean the orElseThrow? Or what exactly?

Reply:
Just the constant try to find the user, have an orElseThrow in there and then instantly catch the exception to throw a different exception. Don't know, it seems weird to me but it could also just be Java being Java

Reply:
Well, it's ok, but you actually don't need to catch it to throw something new.
Don't recall what card it was, but there's one that points to some exception handling documentation, and you could easily handle all these manually thrown exceptions in a single file to actually throw the ResponseStatusException. To the extent of this PR, I think what you did is ok.